{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make final supplementary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "setwd(\"/home/paola/2017/anotations/\")\n",
    "library (stringr)\n",
    "dir=\"/frazer01/projects/CARDIPS/analysis/family1070/ase_chips_v3/analysis\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "###### Closest Gene Annotation\n",
    "gi=read.table(\"/frazer01/publicdata/gencode_v19_20151104/gene_info.tsv\", header=T)\n",
    "pro=c(\"/home/paola/BedFiles/NKXclosestGene.txt\",\"/home/paola/BedFiles/SRFclosestGene.txt\",\n",
    "      \"/home/paola/BedFiles/H3KclosestGene.txt\",\"/home/paola/BedFiles/H3KIPSclosestGene.txt\")\n",
    "\n",
    "##### Motif disruption calculated here ####to be UPDATED######\n",
    "\n",
    "motR=c('/home/paola/Motif_AnalysisR/20_bp/PV.05_NKX/Mix_mot/AlleleAlteringSites_scores_1.txt',\n",
    "        '/home/paola/Motif_AnalysisR/20_bp/PV.05_SRF/Mix_mot/AlleleAlteringSites_scores_1.txt')\n",
    "\n",
    "### Haploreg annotations\n",
    "\n",
    "HAP=c(\"NKX_haploreg.tab\", \"SRF_haploreg.tab\", \"H3K_CM_haploreg.tab\",\"H3K_iPS_haploreg.tab\")\n",
    "QTL=c(\"NKX_eQTLS.txt\" ,\"SRF_eQTLS.txt\", \"H3K_CM_eQTLS.txt\", \"H3K_iPS_eQTLS.txt\")\n",
    "\n",
    "#### Gwas annotations\n",
    "gwasCAT<- read.csv('/home/paola/2017/anotations/GWAS/GwasForFinalAnnotation.csv', header=T)\n",
    "\n",
    "\n",
    "##### left ventricle expression annot\n",
    "lv<-read.csv('/home/paola/Annotations/GTEx Portal.csv', header=T)\n",
    "aa<-read.csv('/home/paola/Annotations/GTEx Portal AA.csv', header=T)\n",
    "\n",
    "lv_eqtls<-read.table(\"/publicdata/gtex_v6/Heart_Left_Ventricle_Analysis.snpgenes\",  sep=\"\\t\", header=T)\n",
    "lv_eqtls$snpID<-paste(\"chr\",lv_eqtls$snp_chrom, \":\", lv_eqtls$snp_pos, sep=\"\")\n",
    "lv<-merge(lv, lv_eqtls[,c('gene', 'snpID' )], by.x='Gencode.Id', by.y='gene')\n",
    "\n",
    "aa_eqtls<-read.table(\"/publicdata/gtex_v6/Heart_Atrial_Appendage_Analysis.snpgenes\",  sep=\"\\t\", header=T)\n",
    "aa_eqtls$snpID<-paste(\"chr\",aa_eqtls$snp_chrom, \":\", aa_eqtls$snp_pos, sep=\"\")\n",
    "aa<-merge(aa, aa_eqtls[,c('gene', 'snpID' )], by.x='Gencode.Id', by.y='gene')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NKX \n",
      "SRF \n",
      "H3K_CM \n"
     ]
    }
   ],
   "source": [
    "nm=c(\"NKX\", \"SRF\", \"H3K_CM\", \"H3K_iPS\")\n",
    "m=data.frame()\n",
    "for (i in 1:3){\n",
    "  cat(nm[i], \"\\n\")\n",
    "  ase=read.csv(paste(dir, \"/HaploregOverlap/SummaryAnnotations.\", nm[i], \".csv\", sep=\"\"), header=T)\n",
    "  ase=subset(ase, ase$combined_pv<0.005)\n",
    "  \n",
    "  \n",
    "  hap=read.table(paste(dir, 'HaploregOverlap', HAP[i], sep=\"/\"), header=F, sep=\"\\t\")\n",
    "  colnames(hap)=c(\"chrom\", \"pos\", \"id\", \"ref\", \"alt\", \"cons\", \"motif\", \"gene\",\"chromHMM\", \"DNAse\")\n",
    "  hap=hap[,c( \"id\", \"motif\", \"gene\",\"chromHMM\", \"DNAse\")]\n",
    "  ase=merge(ase,hap,by.x=\"snpID\", by.y=\"id\", all.x=T)\n",
    " \n",
    "  qtl<-read.table(paste(dir,'GtexOverlap', QTL[i], sep=\"/\"), fill=T, stringsAsFactors=F)\n",
    "  colnames(qtl)<-c(\"snp\",\"qtl\")\n",
    "  ase=merge(ase,qtl,by.x=\"snpID\", by.y=\"snp\", all.x=T)\n",
    "  \n",
    "  gene=read.table(pro[i])\n",
    "  colnames(gene)=c(\"chr\", \"start\", \"end\", \"locus\", \"chr2\", \n",
    "                   \"start2\", \"end2\", \"gene\", \"dot\", \"strand\", \"dist\")\n",
    "\n",
    "  gene=merge(gene, gi, by.x=\"gene\", by.y=\"gene_id\")\n",
    "  gene$closestGenes=paste(gene$gene, gene$gene_name, gene$dist, sep=\"|\")\n",
    "  gene=gene[,c(\"locus\", \"closestGenes\")]\n",
    "  gene=aggregate(closestGenes~locus, gene, paste, collapse=\",\")\n",
    "  ase=merge(ase,gene,by=\"locus\", all.x=T)\n",
    "  \n",
    "  gwasCAT$anno<-paste(gwasCAT$study,gwasCAT$LDsnp,gwasCAT$r2_pop.x, sep='|')\n",
    "  lead<-subset(gwasCAT, !duplicated(gwasCAT[,c(\"SNP_ID_CURRENT\", \"study\")]))\n",
    "  lead<-aggregate(study~SNP_ID_CURRENT, lead, paste, collapse=\";\")\n",
    "  ld<-subset(gwasCAT, !duplicated(gwasCAT[,c(\"LDsnp\", \"anno\")]))\n",
    "  ld<-aggregate(anno~LDsnp, ld, paste, collapse=\";\")\n",
    "  \n",
    "  ase<-merge(ase, lead[, c(\"SNP_ID_CURRENT\", \"study\")], by.x=\"snpID\", by.y=\"SNP_ID_CURRENT\", all.x=T)\n",
    "  ase<-merge(ase, ld[, c(\"LDsnp\", \"anno\")], by.x=\"snpID\", by.y=\"LDsnp\", all.x=T)\n",
    "  \n",
    "\n",
    "  \n",
    "  if (i<3){\n",
    "  mymot=read.table(motR[i], header=T,stringsAsFactor=F)\n",
    "  mymot2=subset(mymot, mymot$maxRefScore-mymot$maxAltScore>0)\n",
    "  \n",
    "  if (i==1){ase$motDisrHomer<-ase$snpID %in% mymot2[mymot2$Motif==\"Nkx2.5_0.93\",4 ]}\n",
    "  if (i==2){ase$motDisrHomer<-ase$snpID %in% mymot2[mymot2$Motif==\"CTCF_Boris\",4 ]}\n",
    "  \n",
    "  mymot[is.na(mymot)]<-0\n",
    "  mymot=merge(mymot, ase[,c('variantID', 'new_maf')], by.x=\"SNP\", by.y=\"variantID\")\n",
    "  mymot[mymot$ref_maf>0.5,'scoreRef']<-mymot[mymot$ref_maf>0.5,c('maxRefScore')]\n",
    "  mymot[mymot$ref_maf<0.5,'scoreRef']<-mymot[mymot$ref_maf<0.5,c('maxAltScore')]\n",
    "  mymot[mymot$ref_maf>0.5,'scoreAlt']<-mymot[mymot$ref_maf>0.5,c('maxAltScore')]\n",
    "  mymot[mymot$ref_maf<0.5,'scoreAlt']<-mymot[mymot$ref_maf<0.5,c('maxRefScore')]\n",
    "  mymot$Diff<-round(mymot$scoreAlt-mymot$scoreRef,2)\n",
    "  \n",
    "  mymot$Motif_anno<-paste(mymot$Motif, mymot$Diff,mymot$maxRefScoreStrand, \n",
    "                          round(mymot$scoreRef,2), round(mymot$scoreAlt,2), sep=\"|\" )\n",
    "  \n",
    "  agg=aggregate(Motif_anno~SNP, mymot, paste, collapse=\",\")\n",
    "  ase=merge(ase, agg, by.x=\"variantID\", by.y=\"SNP\", all.x=T)\n",
    "  #ase=subset(ase, select=-c(anyOthMot, coreMot))\n",
    "  }\n",
    "  \n",
    "  \n",
    "  if (i>2){\n",
    "    ase$motDisrHomer<-NA\n",
    "    ase$Motif_anno<-NA\n",
    "    #ase$motDisr<-NA\n",
    "    #ase$scoreDiff<-NA\n",
    "    ase<-subset(ase, select=colnames(m))\n",
    "  }\n",
    "\n",
    " \n",
    "  m<-rbind(m,ase)\n",
    "  \n",
    "}\n",
    "\n",
    "m<-subset(m,m$combined_fdr<0.05)\n",
    "\n",
    "m$topxp<-m$snpID %in% lv$snpID | m$snpID %in% aa$snpID\n",
    "m$bestp<-m$snpID %in% lv_eqtls$snpID[lv_eqtls$has_best_p==1] | \n",
    "  m$snpID %in% aa_eqtls$snpID[aa_eqtls$has_best_p==1]\n",
    "\n",
    "\n",
    "write.table(m, \"Supplementary_table_ase_annotations.txt\", sep=\"\\t\", row.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "m<-subset(m,m$combined_fdr<0.05)\n",
    "\n",
    "m$topxp<-m$snpID %in% lv$snpID | m$snpID %in% aa$snpID\n",
    "m$bestp<-m$snpID %in% lv_eqtls$snpID[lv_eqtls$has_best_p==1] | \n",
    "  m$snpID %in% aa_eqtls$snpID[aa_eqtls$has_best_p==1]\n",
    "\n",
    "write.table(m, \"Supplementary_table_ase_annotations.txt\", sep=\"\\t\", row.names=FALSE, quote=FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'variantID'</li>\n",
       "\t<li>'snpID'</li>\n",
       "\t<li>'locus'</li>\n",
       "\t<li>'combined_pv'</li>\n",
       "\t<li>'combined_fdr'</li>\n",
       "\t<li>'new_maf'</li>\n",
       "\t<li>'refAllele'</li>\n",
       "\t<li>'altAllele'</li>\n",
       "\t<li>'mark'</li>\n",
       "\t<li>'tissue'</li>\n",
       "\t<li>'cons'</li>\n",
       "\t<li>'dsQTLs'</li>\n",
       "\t<li>'anyGWAS'</li>\n",
       "\t<li>'cvdGWAS'</li>\n",
       "\t<li>'hrtGWAS'</li>\n",
       "\t<li>'motif'</li>\n",
       "\t<li>'gene'</li>\n",
       "\t<li>'chromHMM'</li>\n",
       "\t<li>'DNAse'</li>\n",
       "\t<li>'qtl'</li>\n",
       "\t<li>'closestGenes'</li>\n",
       "\t<li>'study'</li>\n",
       "\t<li>'anno'</li>\n",
       "\t<li>'motDisrHomer'</li>\n",
       "\t<li>'Motif_anno'</li>\n",
       "\t<li>'topxp'</li>\n",
       "\t<li>'bestp'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'variantID'\n",
       "\\item 'snpID'\n",
       "\\item 'locus'\n",
       "\\item 'combined\\_pv'\n",
       "\\item 'combined\\_fdr'\n",
       "\\item 'new\\_maf'\n",
       "\\item 'refAllele'\n",
       "\\item 'altAllele'\n",
       "\\item 'mark'\n",
       "\\item 'tissue'\n",
       "\\item 'cons'\n",
       "\\item 'dsQTLs'\n",
       "\\item 'anyGWAS'\n",
       "\\item 'cvdGWAS'\n",
       "\\item 'hrtGWAS'\n",
       "\\item 'motif'\n",
       "\\item 'gene'\n",
       "\\item 'chromHMM'\n",
       "\\item 'DNAse'\n",
       "\\item 'qtl'\n",
       "\\item 'closestGenes'\n",
       "\\item 'study'\n",
       "\\item 'anno'\n",
       "\\item 'motDisrHomer'\n",
       "\\item 'Motif\\_anno'\n",
       "\\item 'topxp'\n",
       "\\item 'bestp'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'variantID'\n",
       "2. 'snpID'\n",
       "3. 'locus'\n",
       "4. 'combined_pv'\n",
       "5. 'combined_fdr'\n",
       "6. 'new_maf'\n",
       "7. 'refAllele'\n",
       "8. 'altAllele'\n",
       "9. 'mark'\n",
       "10. 'tissue'\n",
       "11. 'cons'\n",
       "12. 'dsQTLs'\n",
       "13. 'anyGWAS'\n",
       "14. 'cvdGWAS'\n",
       "15. 'hrtGWAS'\n",
       "16. 'motif'\n",
       "17. 'gene'\n",
       "18. 'chromHMM'\n",
       "19. 'DNAse'\n",
       "20. 'qtl'\n",
       "21. 'closestGenes'\n",
       "22. 'study'\n",
       "23. 'anno'\n",
       "24. 'motDisrHomer'\n",
       "25. 'Motif_anno'\n",
       "26. 'topxp'\n",
       "27. 'bestp'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"variantID\"    \"snpID\"        \"locus\"        \"combined_pv\"  \"combined_fdr\"\n",
       " [6] \"new_maf\"      \"refAllele\"    \"altAllele\"    \"mark\"         \"tissue\"      \n",
       "[11] \"cons\"         \"dsQTLs\"       \"anyGWAS\"      \"cvdGWAS\"      \"hrtGWAS\"     \n",
       "[16] \"motif\"        \"gene\"         \"chromHMM\"     \"DNAse\"        \"qtl\"         \n",
       "[21] \"closestGenes\" \"study\"        \"anno\"         \"motDisrHomer\" \"Motif_anno\"  \n",
       "[26] \"topxp\"        \"bestp\"       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'snpID'</li>\n",
       "\t<li>'locus'</li>\n",
       "\t<li>'variantID'</li>\n",
       "\t<li>'combined_pv'</li>\n",
       "\t<li>'combined_fdr'</li>\n",
       "\t<li>'new_maf'</li>\n",
       "\t<li>'refAllele'</li>\n",
       "\t<li>'altAllele'</li>\n",
       "\t<li>'mark'</li>\n",
       "\t<li>'tissue'</li>\n",
       "\t<li>'cons'</li>\n",
       "\t<li>'dsQTLs'</li>\n",
       "\t<li>'anyGWAS'</li>\n",
       "\t<li>'cvdGWAS'</li>\n",
       "\t<li>'hrtGWAS'</li>\n",
       "\t<li>'motif'</li>\n",
       "\t<li>'gene'</li>\n",
       "\t<li>'chromHMM'</li>\n",
       "\t<li>'DNAse'</li>\n",
       "\t<li>'qtl'</li>\n",
       "\t<li>'closestGenes'</li>\n",
       "\t<li>'study'</li>\n",
       "\t<li>'anno'</li>\n",
       "\t<li>'motDisrHomer'</li>\n",
       "\t<li>'Motif_anno'</li>\n",
       "\t<li>'motDisr'</li>\n",
       "\t<li>'scoreDiff'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'snpID'\n",
       "\\item 'locus'\n",
       "\\item 'variantID'\n",
       "\\item 'combined\\_pv'\n",
       "\\item 'combined\\_fdr'\n",
       "\\item 'new\\_maf'\n",
       "\\item 'refAllele'\n",
       "\\item 'altAllele'\n",
       "\\item 'mark'\n",
       "\\item 'tissue'\n",
       "\\item 'cons'\n",
       "\\item 'dsQTLs'\n",
       "\\item 'anyGWAS'\n",
       "\\item 'cvdGWAS'\n",
       "\\item 'hrtGWAS'\n",
       "\\item 'motif'\n",
       "\\item 'gene'\n",
       "\\item 'chromHMM'\n",
       "\\item 'DNAse'\n",
       "\\item 'qtl'\n",
       "\\item 'closestGenes'\n",
       "\\item 'study'\n",
       "\\item 'anno'\n",
       "\\item 'motDisrHomer'\n",
       "\\item 'Motif\\_anno'\n",
       "\\item 'motDisr'\n",
       "\\item 'scoreDiff'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'snpID'\n",
       "2. 'locus'\n",
       "3. 'variantID'\n",
       "4. 'combined_pv'\n",
       "5. 'combined_fdr'\n",
       "6. 'new_maf'\n",
       "7. 'refAllele'\n",
       "8. 'altAllele'\n",
       "9. 'mark'\n",
       "10. 'tissue'\n",
       "11. 'cons'\n",
       "12. 'dsQTLs'\n",
       "13. 'anyGWAS'\n",
       "14. 'cvdGWAS'\n",
       "15. 'hrtGWAS'\n",
       "16. 'motif'\n",
       "17. 'gene'\n",
       "18. 'chromHMM'\n",
       "19. 'DNAse'\n",
       "20. 'qtl'\n",
       "21. 'closestGenes'\n",
       "22. 'study'\n",
       "23. 'anno'\n",
       "24. 'motDisrHomer'\n",
       "25. 'Motif_anno'\n",
       "26. 'motDisr'\n",
       "27. 'scoreDiff'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"snpID\"        \"locus\"        \"variantID\"    \"combined_pv\"  \"combined_fdr\"\n",
       " [6] \"new_maf\"      \"refAllele\"    \"altAllele\"    \"mark\"         \"tissue\"      \n",
       "[11] \"cons\"         \"dsQTLs\"       \"anyGWAS\"      \"cvdGWAS\"      \"hrtGWAS\"     \n",
       "[16] \"motif\"        \"gene\"         \"chromHMM\"     \"DNAse\"        \"qtl\"         \n",
       "[21] \"closestGenes\" \"study\"        \"anno\"         \"motDisrHomer\" \"Motif_anno\"  \n",
       "[26] \"motDisr\"      \"scoreDiff\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(m)\n",
    "colnames(ase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.2.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
